oak-core/src/test/java/org/apache/jackrabbit/oak/query/AbstractQueryTest.java

```diff
@@ -227,7 +228,16 @@ protected void test(String file) throws Exception {
                     w.println(line);
                     line = line.substring("commit".length()).trim();
                     apply(root, line);
-                    root.commit();
+                    // This part of code is used by both lucene and elastic tests.
+                    // The index definitions in these tests don't have async property set
+                    // So lucene, in this case behaves in a sync manner. But the tests fail on Elastic,
+                    // since ES indexing is always async.
+                    // The below commit info map (sync-xmode = rt) would make Elastic use RealTimeBulkProcessHandler.
```

Comment:
another alternative would be to create a method (eg `syncIndex`) in `AbstractQueryTest` overridden in `ElasticAbstractQueryTest` to pass the `sync-mode` option.

Reply:
Leaving it in AbstractQueryTest for now, since the additional commit info doesn't really impact lucene.